sqlalchemy_redshift/commands.py

```diff
@@ -18,10 +18,21 @@
 ACCESS_KEY_ID_RE = re.compile('[A-Z0-9]{20}')
 SECRET_ACCESS_KEY_RE = re.compile('[A-Za-z0-9/+=]{40}')
 TOKEN_RE = re.compile('[A-Za-z0-9/+=]+')
+IAM_ROLE_RE = re.compile('arn:aws:iam::[0-9]+:role/.+')
 
 
-def _process_aws_credentials(access_key_id, secret_access_key,
-                             session_token=None):
+def _process_aws_credentials(access_key_id=None, secret_access_key=None,
+                             session_token=None, iam_role=None):
+
+    if iam_role is not None:
+        if not IAM_ROLE_RE.match(iam_role):
+            raise ValueError(
+                'invalid IAM role ARN; does not match {pattern}'.format(
+                    pattern=IAM_ROLE_RE.pattern,
+                )
+            )
+        else:
+            return 'aws_iam_role=' + iam_role
```

Comment:
@mtrbean can you check split the iam_role into aws_account_id and iam_role_name validating both against a regex.

```python
if iam_role_name is not None:
    if not AWS_ACCOUNT_ID_RE.match(aws_account_id):
        raise ValueError(...)
    if not IAM_ROLE_NAME_RE.match(iam_role_Name):
        raise ValueError(...)

    credentials = 'aws_iam_role=arn:aws:iam::{aws_acount_id}:role/{iam_role_name}'.format(...)
```